src/bentoml/_internal/service/service.py

```diff
@@ -226,6 +230,47 @@ def __init__(
             models=[] if models is None else models,
         )
 
+        # Set import origin info - import_str can not be determined at this stage yet as
+        # the variable name is only available in module vars after __init__ is returned
+        # get_service_import_origin below will use the _caller_module for retriving the
+        # correct import_str for this service
+        caller_module = inspect.currentframe().f_back.f_globals["__name__"]
+        object.__setattr__(self, "_caller_module", caller_module)
+        object.__setattr__(self, "_working_dir", os.getcwd())
+
+    def get_service_import_origin(self) -> tuple[str, str]:
+        """
+        Returns the module name and working directory of the service
+        """
+        if not self._import_str:
+            import_module = self._caller_module
+            if import_module == "__main__":
+                if hasattr(sys.modules["__main__"], "__file__"):
+                    import_module = sys.modules["__main__"].__file__
+                else:
+                    raise BentoMLException(
+                        "Failed to get service import origin, bentoml.Service object defined interactively in console or notebook is not supported"
+                    )
+
+            for name, value in vars(sys.modules[self._caller_module]).items():
+                if value is self:
+                    object.__setattr__(self, "_import_str", f"{import_module}:{name}")
+                    break
+
+            if not self._import_str:
+                raise BentoMLException(
+                    "Failed to get service import origin, bentoml.Service object must be assigned to a variable at module level"
```

Comment:
nit:

```suggestion
                    "Failed to get service import origin, bentoml.Service object must be loaded eagerly (it should be available in module scope)."
```

Reply:
`must be loaded eagerly (it should be available in module scope)."` is not technically correct here. To simplify put, it should be defined and assigned at the module level (not in/under module scope, that's a bit different..)